document/CoreFormat.mdk

```diff
@@ -826,12 +791,111 @@ In this case, it is helpful to the author of the logging component to know
+the tool executable, that information should be redacted or omitted.
+Redacted information should be replaced with the token `[REMOVED]`.
+
+~ IsoExample
+Suppose a tool is invoked with the command line
+
+```
+    C:\Users\johnsmith\Tools\DbScanner\DbScanner.exe
+        /ConnectionString "Server=CorpServer;Database=Accounting;User Id=Admin;Password=S3cr#t" /input *.sql
+```
+
+Then the value of the `commandLine` property might contain the redacted command line
+
+```
+    [REMOVED]\DbScanner.exe /connectionString=[REMOVED] /input=*.sql 
+```
+~
+
+~ IsoNote
+Redacting sensitive information from `commandLine` makes it more difficult to precisely reproduce
+an analysis run. The value of `commandLine` would have to be combined with information from another
+source to allow the run to be repeated.
+~
+
+### `machine` property { #invocation-machine }
+
+An `invocation` object may contain a property named `machine` whose value is a string containing
+the name of the machine on which the tool was run.
+
+### `account` property { #invocation-account }
+
+An `invocation` object may contain a property named `account` whose value is a string containing
+the name of the account under which the tool was run.
+
+### `processId` property { #invocation-processId }
+
+An `invocation` object may contain a property named `processId` whose value is an integer containing
+the id of the process in which the tool was run.
+
+### `fileName` property { #invocation-fileName }
+
+An `invocation` object may contain a property named `fileName` whose value is a string containing
+the fully qualified path name of the tool's executable file.
+
+~ IsoNumberedNote { @IsoNumberedNote=1 }
+This property is defined in the `invocation` object rather than in the `tool` object (&sect;[#tool]) because the
+identical tool might be invoked from different paths on different machines.
+~
+
+~ IsoNumberedNote
+This property might duplicate information in the `commandLine` property (&sect;[#invocation-commandLine]).
+Storing the file name in a separate property eliminates the need to parse the command line to isolate
+the file name.
+~
+
+~ IsoNumberedNote
+Absolute path names can reveal information that might be sensitive.
+~
+
+### `workingDirectory` property { #invocation-workingDirectory }
+
+An `invocation` object may contain a property named `workingDirectory` whose value is a string containing
+the fully qualified path name of the directory in which the analysis tool was invoked.
+
+~ IsoNote
+Absolute path names can reveal information that might be sensitive.
+~
+
+### `environmentVariables` property { #invocation-environmentVariables }
+
+An `invocation` object may contain a property named `environmentVariables` whose value is an object.
+The property names in this object contain the names of the environment variables in the tool's
+execution environment. The value of each property is a string containing the value of the specified
+environment variable.
```

Comment:
Need a note that this information is highly sensitive. Environment variables on a windows machine includes user data, machine name, %PATH% variable, logon domain controller, etc.